icu4j/main/classes/core/src/com/ibm/icu/text/DecimalFormatSymbols.java

```diff
@@ -888,9 +888,32 @@ public void setCurrency(Currency currency) {
         if (currency == null) {
             throw new NullPointerException();
         }
+        if (currency.equals(this.currency)) {
+            return;
+        }
+        CurrencyDisplayInfo displayInfo = CurrencyData.provider.getInstance(ulocale, true);
+        setCurrencyOrNull(currency, displayInfo);
+    }
+
+    private void setCurrencyOrNull(Currency currency, CurrencyDisplayInfo displayInfo) {
         this.currency = currency;
+
+        if (currency == null) {
```

Comment:
Should we have a test for the case when currency == null?  Or documentation of when such a case would actually occur?

Reply:
This code path does already get exercised 128 times in NumberFormatTest; basically whenever a DecimalFormat is instantiated with a locale without a region (like "en" instead of "en-US").  So I'd say that test coverage is already sufficient.

As far as documentation goes, the behavior should be unchanged from before.  Previously, these fallback symbols were set in the "initialize" method; now they're being set in this method.